Content.Shared/Animals/ShearableSystem.cs

```diff
@@ -0,0 +1,395 @@
+    /// <returns>
+    ///     A <c>ShearableComponent.CheckShearReturns</c> enum of the result.
+    /// </returns>
+    /// <seealso cref="CheckShearReturns"/>
+    public CheckShearReturns CheckShear(Entity<ShearableComponent> ent, out EntityPrototype shearedProduct, EntityUid? usedItem = null, bool checkItem = true)
+    {
+        return CheckShear(ent, out shearedProduct, out _, out _, out _, usedItem, checkItem);
+    }
+
+    /// <summary>
+    ///     Checks if the target entity can currently be sheared. Overload function that requires no out variables.
+    /// </summary>
+    /// <param name="ent">The shearable entity that will be checked, and the ShearableComponent combined with it.</param>
+    /// <param name="usedItem">The held item that is being used to shear the target entity.</param>
+    /// <param name="checkItem">If false then skip checking for the correct shearing tool.</param>
+    /// <returns>
+    ///     A <c>ShearableComponent.CheckShearReturns</c> enum of the result.
+    /// </returns>
+    /// <seealso cref="CheckShearReturns"/>
+    public CheckShearReturns CheckShear(Entity<ShearableComponent> ent, EntityUid? usedItem = null, bool checkItem = true)
+    {
+        return CheckShear(ent, out _, out _, out _, out _, usedItem, checkItem);
+    }
+
+    /// <summary>
+    ///     Handles shearing when the player left-clicks an entity.
+    ///     Doesn't run any checks, those are handled by AttemptShear.
+    /// </summary>
+    private void OnInteractUsingEvent(Entity<ShearableComponent> ent, ref InteractUsingEvent args)
+    {
+        // If no tool is specified then this might take over empty-hand interaction of an entity.
+        // But, sheep have a default petting action so presumably this can also be overidden up the hierarchy.
+        // All checks run from AttemptShear.
+        AttemptShear(ent, args.User, args.Used);
+    }
+
+    /// <summary>
+    ///     Attempts to shear the target animal, checking if it is shearable and building arguments for calling TryStartDoAfter.
+    ///     Called by the "shear" verb.
+    /// </summary>
+    private void AttemptShear(Entity<ShearableComponent> ent, EntityUid userUid, EntityUid? toolUsed)
+    {
+        // Run all shearing checks.
+        switch (CheckShear(ent, out var shearedProduct, toolUsed))
+        {
+            case CheckShearReturns.Success:
+                // ALL SYSTEMS GO!
+                break;
+            case CheckShearReturns.WrongTool:
+                // No message because it would cause excessive popups when e.g. you're feeding the sheep some food.
+                return;
+            case CheckShearReturns.InsufficientSolution:
+                // NO WOOL LEFT.
+                _popup.PopupClient(Loc.GetString("shearable-system-no-product", ("target", Identity.Entity(ent.Owner, EntityManager)), ("product", shearedProduct.Name)), ent.Owner, userUid);
+                return;
```

Comment:
```suggestion
        if (!CheckShear(ent, out var shearedProduct, toolUsed, out var string))
        {
            if (string != null)
                _popup.PopupClient(string, ent.Owner, userUid);
            return;
        }
```

Do this and then just handle the Loc stuff inside CheckShear, rename it to CanShear for consistency.